pkg/controller/instance/instance_controller.go

```diff
@@ -189,15 +190,15 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 	return reconcile.Result{}, nil
 }
 
-func preparePlanExecution(instance *kudov1alpha1.Instance, ov *kudov1alpha1.OperatorVersion, activePlanStatus *kudov1alpha1.PlanStatus) (*activePlan, *engineMetadata, error) {
+func preparePlanExecution(instance *kudov1alpha1.Instance, ov *kudov1alpha1.OperatorVersion, activePlanStatus *kudov1alpha1.PlanStatus) (*activePlan, *task.EngineMetadata, error) {
```

Comment:
I think the name is somewhat misleading. It's more of `TaskExecutionMetadata` actually. But anyway - I **had** to move it down here because you get circular dependencies otherwise  ¯\_(ツ)_/¯